spec/controllers/chapters_controller_spec.rb

```diff
@@ -473,6 +473,15 @@
             expect(assigns[:chapter].posted).to be true
           end
 
+          it "updates cached chapter counts" do
+            work.invalidate_work_chapter_count(work)
+            expect(work.number_of_chapters).to eq 1
+            expect(work.number_of_posted_chapters).to eq 1
+            post :create, params: { work_id: work.id, chapter: chapter_attributes, post_without_preview_button: true }
+            expect(work.number_of_chapters).to eq 2
+            expect(work.number_of_posted_chapters).to eq 2
```

Comment:
Since you're checking the same fields before and after an action, you can use the change matcher, like this:

https://github.com/otwcode/otwarchive/blob/1f8558452e6984a41ace47074324013d2a3454f8/spec/controllers/admin/admin_users_controller_spec.rb#L109-L122

`.from` and `.to` should be short enough to live on the same line as the `change` block in this case.

Reply:
done in [c9b6691](https://github.com/otwcode/otwarchive/pull/4159/commits/c9b66913424ce335e9aacdf475005d93aed55a09)